src/client/mod.rs

```diff
@@ -687,6 +691,14 @@ impl Client {
     /// [gateway docs]: crate::gateway#sharding
     #[instrument(skip(self))]
     pub async fn start(&mut self) -> Result<()> {
+        #[cfg(feature = "unstable_discord_api")]
+        {
+            if &self.application_id.is_none() {
+                let application_id =
+                    parse_token(&self.token).expect("Couldn't parse token correctly").bot_user_id;
+                self.application_id(application_id);
+            }
+        }
```

Comment:
Why not do this in the instantiation of the `Client` in the `ClientBuilder`'s `Future` implementation? The `Client` has other `start` methods. Doing this only here would introduce an inconsistency.

Reply:
I guess that makes sense, just means that there wouldn't be any doc comments.